tests/test_slam_service.py

```diff
@@ -0,0 +1,137 @@
+from typing import List
+import pytest
+
+from grpclib.testing import ChannelFor
+
+from viam.proto.common import DoCommandRequest, DoCommandResponse
+from viam.proto.service.slam import (
+    GetInternalStateRequest,
+    GetInternalStateResponse,
+    GetPointCloudMapRequest,
+    GetPointCloudMapResponse,
+    GetPositionRequest,
+    GetPositionResponse,
+    SLAMServiceStub,
+)
+from viam.resource.manager import ResourceManager
+from viam.services.slam import Pose, SLAMServiceClient, SLAMServiceRPCService
+from viam.utils import dict_to_struct, struct_to_dict
+
+from .mocks.services import MockSLAMService
+
+INTERNAL_STATE_CHUNKS = [bytes(5), bytes(2)]
```

Comment:
(q) There's some repetition here with `tests/mocks/services.py` and `examples/servers/v1/services.py`. I assume there's a good reason for the duplication at least as far as the example code goes, but is it possible/would it make sense to dedup in the `tests/` code?

Reply:
I know that they're used for different things. I followed the way we do components for services. We also do the same for `modules`. As for why we split them this way, I'm not entirely sure.